include/phasar/PhasarLLVM/DataFlowSolver/IfdsIde/Problems/IDEInstInteractionAnalysis.h

```diff
@@ -205,16 +205,14 @@ class IDEInstInteractionAnalysisT
       std::variant<n_t, const llvm::GlobalVariable *> InstOrGlobal);
 
   IDEInstInteractionAnalysisT(
-      const ProjectIRDB *IRDB, const LLVMTypeHierarchy *TH,
-      const LLVMBasedICFG *ICF, LLVMPointsToInfo *PT,
+      const ProjectIRDB *IRDB, const LLVMBasedICFG *ICF, LLVMPointsToInfo *PT,
```

Comment:
Do I understand this correctly, `TH` now is no longer needed on the Problem by the Solver takes care of this?

And for the `ICFG` and `PT` we store the locally?

Reply:
Yes, in fact TH is not used at all by the analysis problems, so there is no need to them to store a reference to it. As of now, the only use of TH is in the call-graph resolver inside LLVMBasedICFG.. The PT and ICFG are used in some analysis problems, but by far not all, so only those problems needing them will recieve a pointer. Previously, all analysis problems needed to store all of TH, PT and ICFG -- and then not using them. In future, we might want to only compute that subset of {TH, PT, ICFG} that is really needed.

The solver only needs the problem and the ICFG, nothing more. Previously, the solver retrieved its ICFG from the problem; now it gets it directly as the problem may not have access to an ICFG. Inside the solver, there is no difference at all, as it directly stores a pointer to the ICFG anyway.

Reply:
LGTM